.github/workflows/build_canary.yml

```
@@ -13,61 +13,114 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      - name: Build and push operator image
-        uses: docker/build-push-action@v1.1.1
+      - name: Set up tags and refs
+        id: prep
+        run: |
+          SHA="sha-${GITHUB_SHA::8}"
+          echo ::set-output name=sha::${SHA}
+          echo ::set-output name=created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+
+      - name: Set up Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to GHCR
+        uses: docker/login-action@v1
         with:
           # Username used to log in to a Docker registry. If not set then no login will occur
-          username: ${{ secrets.GHCR_AUTH_USER }}
+          username: ${{ github.repository_owner }}
           # Password or personal access token used to log in to a Docker registry. If not set then no login will occur
           password: ${{ secrets.GHCR_AUTH_PAT }}
           # Server address of Docker registry. If not set then will default to Docker Hub
           registry: ghcr.io
+
+      - name: Build and push operator image
+        uses: docker/build-push-action@v2
+        with:
           # Docker repository to tag the image with
-          repository: ${{ secrets.GHCR_AUTH_USER }}/http-add-on-operator
-          # Automatically tags the built image with the git SHA
-          tag_with_sha: true
-          tags: canary
-          dockerfile: operator/Dockerfile
+          repository: ${{ github.repository_owner }}/http-add-on-operator
+          tags: canary,${{ steps.prep.outputs.sha}}
+          labels: |
+            sh.keda.http.image.source=${{github.event.repository.html_url}}
+            sh.keda.http.image.created=${{steps.prep.outputs.created}}
+            sh.keda.http.image.revision=${{github.sha}}
+          file: operator/Dockerfile
+          context: operator
```

We need to use `.` as the context because the `Dockerfile`s need to bring the `go.mod`, `go.sum`, and some other stuff into the image

```suggestion
          context: .
```